gomod/module.go

```diff
@@ -0,0 +1,100 @@
+package gomod
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+)
+
+// InferModuleVersion determines the go module version that is supplied by the
+// code in the given project root.
+func InferModuleVersion(projectRoot string) (string, error) {
```

Comment:
Just a note, not 100% sure if this function aware/handles this: Go modules does not allow v2.x.x tag for module path without containing a `/v2`, similar for `v3`, `v4` etc.

Eg. `module github.com/sourcegraph/sourcegraph` is only allowed to use tag v0 and up to v1. If try to use `v2` the Go modules mode won't allow user to compile. `module github.com/sourcegraph/sourcegraph/v2` is the module path Go modules toolchain expects to see when using tags of v2.x.x.

Even if we publish `module github.com/sourcegraph/sourcegraph` starting at v2 (no v1 or v0 whatsoever), it won't work AFAIK.

PS: Modules has import path `gopkg.in/xxx` are special cases that `xxx.v2` could work (not `xxx/v2`).

Reply:
If we haven't handled this, I don't think we have to take care of it now, assumes users would have fixed the compile issue before pushing to the code host.

Reply:
This is good info. Could you make an issue so that we remember to handle this at some point?